src/eredis_cluster.erl

```diff
@@ -495,9 +496,11 @@ update_hash_field(Key, Field, UpdateFunction) ->
 %% http://redis.io/topics/transactions
 %% @end
 %% =============================================================================
--spec optimistic_locking_transaction(WatchedKey::anystring(), GetCommand::redis_command(),
-    UpdateFunction::fun((redis_result()) -> redis_pipeline_command())) ->
-        {redis_transaction_result(), any()}.
+-spec optimistic_locking_transaction(WatchedKey :: anystring(),
+                                     GetCommand :: redis_command(),
+                                     UpdateFunction :: fun((redis_result()) -> redis_pipeline_command())) ->
+          {ok, {redis_success_result(), any()}} | {ok, {[redis_success_result()], any()}}
+          | optimistic_locking_error_result().
```

Comment:
The return type can be replaced by `optimistic_locking_result()` I think.

Reply:
unfortunately not, it changes the internal format needed by transaction to another that is returned from this function..

Reply:
Ok, it's slightly different, I see now.